### DISCUSSION TITLE
How to get full length of tweets?

### DISCUSSION THREAD

**Discussion by tobiashill** [redouane59/twittered#128] [Q&A] [answered]
Hi, 

I like the twittered-lib but I cannot get full length tweets as a result. They are truncated after some 140 characters. 

Other libs provide means to send the tweet_mode=extended flag (see discussion here for instance: https://stackoverflow.com/questions/48056010/how-to-get-the-full-tweet-of-the-user-timeline-with-twitter4j). 

So, is this possible in twittered? How to do it?

**Comment by tobiashill** [comment]
Nevermind ... my bad ... it seems to be full length as long as it is not a RT.

**Reply by redouane59** [reply]
Hi, do you have an example of truncated retweet ?

**Reply by gabac** [reply]
Perfect timing. Looking at the same issue. Here is an example of a tweet: `1358782390089494531`
https://twitter.com/VMwareTanzuDACH/status/1358782390089494531

I fetched the tweet via the `favourites` endpoint
```
"text":"Die @oop_conference startet heute! Verpasst nicht eines unserer Speaker-Highlights: \"Cloud Native Java\" mit… https://t.co/E6tITmeuHo"
```

**Reply by redouane59** [reply]
Wow well seen I didn't know that ! I'm going to check what parameter is missing to retreive everything
https://developer.twitter.com/en/docs/twitter-api/tweets/lookup/api-reference/get-tweets

**Reply by gabac** [reply]
Setting the param `tweet_mode` to `extended` returns everything
https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/follow-search-get-users/api-reference/get-users-lookup

**Reply by redouane59** [reply]
I'm using V2 of endpoint not V1, see https://developer.twitter.com/en/docs/twitter-api/tweets/lookup/api-reference/get-tweets

**Reply by gabac** [reply]
Correct but this param is only present in the V1 endpoint. V2 has no truncated tweets. My example from above is also from a V1 endpoint

**Reply by redouane59** [reply]
Can you precise me which method of the library your are using ?  Is it `getFavorites()` from `ITwitterClientV1` class ?
As some endpoints don't exist yet in V2 it's probable that a V1 call has to be changed somewhere in the code.

**Reply by gabac** [reply]
Yes I'm using the method `getFavorites()` from `ITwitterClientV1` as this endpoint, as you mentioned, does not yet exist in V2.
The param `tweet_mode` only applies to *V1* and can be applied to any endpoint, e.g. 
`https://api.twitter.com/1.1/favorites/list.json?count=200&screen_name=gabac&tweet_mode=extended`
which makes sure that tweets are no longer truncated.

It would make sense to offer this on a top level so it gets applied to all of the V1 endpoints if desired.

**Reply by redouane59** [reply]
Perfect I added the info into https://github.com/redouane59/twittered/issues/130 , will check that for the next release, thank you very much guys !

**Reply by gabac** [reply]
See for example Twitter4J
https://github.com/Twitter4J/Twitter4J/blob/8376fade8d557896bb9319fb46e39a55b134b166/twitter4j-core/src/main/java/twitter4j/TwitterImpl.java#L75

**Comment by redouane59** [comment, marked as answer]
Is fixed thanks to PR https://github.com/redouane59/twittered/pull/133 , will be available on the next release 1.19
